include/zcbor_debug.h

```diff
@@ -62,6 +62,43 @@ static void zcbor_print_compare_strings_diff(const uint8_t *str1, const uint8_t
 	}
 }
 
+__attribute__((used))
+static const char *zcbor_error_str(int error)
+{
+	#define ZCBOR_ERR_CASE(err) case err: { \
+			static const char err_str[] = #err; \
```

Comment:
I disagree. AFAIK, if returned directly, we have no guarantee that the memory has static lifetime, and even if we do, I assume the generated assembly will be identical to the current solution. I'd be interested to see it if you have some documentation on this though.

Reply:
[C99 standard](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf) *6.4.5 String Literals* says this:
> In translation phase 7, a byte or code of value zero is appended to each multibyte
character sequence that results from a string literal or literals.66) The multibyte character
sequence is then used to initialize an array of **static storage** duration and length just
sufficient to contain the sequence.

I had been confused the first time I saw `return "string";`, because I had been sure there wasn't any guarantee that the string would still live once the function returns. However, what I conclude from the spec quoted above, they are `static` regardless if they're declared `static` or not.

Reply:
That's super interesting, didn't know that! Changed it now :).

Reply:
I was surprised as well :D